custom/icds/tasks/__init__.py

```diff
@@ -31,25 +37,23 @@ def _get_query(db_name, max_age=max_age):
             created_on__lt=max_age
         ).order_by('created_on')
 
-    run_again = False
-    for db_name in get_db_aliases_for_partitioned_query():
-        bytes_deleted = 0
-        query = _get_query(db_name)
-        metas = list(query[:1000])
-        if metas:
-            for meta in metas:
-                bytes_deleted += meta.content_length or 0
-            db.bulk_delete(metas=metas)
-
-            tags = [f'database:{db_name}']
-            age = datetime.utcnow() - metas[-1].created_on
-            datadog_gauge('commcare.icds_images.max_age', value=age.total_seconds(), tags=tags)
-            row_estimate = estimate_row_count(query, db_name)
-            datadog_gauge('commcare.icds_images.count_estimate', value=row_estimate, tags=tags)
-            datadog_counter('commcare.icds_images.bytes_deleted', value=bytes_deleted)
-            datadog_counter('commcare.icds_images.count_deleted', value=len(metas))
-            run_again = True
+    bytes_deleted = 0
+    query = _get_query(db_name)
+    metas = list(query[:1000])
+    run_again = len(metas) == 1000
+    if metas:
+        for meta in metas:
+            bytes_deleted += meta.content_length or 0
+        db.bulk_delete(metas=metas)
+
+        tags = [f'database:{db_name}']
+        age = datetime.utcnow() - metas[-1].created_on
+        datadog_gauge('commcare.icds_images.max_age', value=age.total_seconds(), tags=tags)
+        row_estimate = estimate_row_count(query, db_name)
+        datadog_gauge('commcare.icds_images.count_estimate', value=row_estimate, tags=tags)
+        datadog_counter('commcare.icds_images.bytes_deleted', value=bytes_deleted)
+        datadog_counter('commcare.icds_images.count_deleted', value=len(metas))
 
     runtime = datetime.utcnow() - cutoff
     if run_again and runtime.total_seconds() < MAX_RUNTIME:
-        delete_old_images.delay(cutoff)
+        delete_old_images_on_db.delay(cutoff)
```

Comment:
Doesn't this also need to pass `db_name`?

Reply:
Thanks